frontend/src/store/index.ts

```diff
@@ -1,20 +1,59 @@
 import Vue from "vue";
 import Vuex from "vuex";
+import axios from "axios";
+import { AxiosInstance } from "axios";
 
 Vue.use(Vuex);
 
+interface State {
+  isAuthenticated: boolean;
+  username: string | undefined;
+  avatar: string | undefined | null;
+  id: string | undefined;
+  token: string | undefined;
+}
+
+const state: State = {
+  isAuthenticated: false,
+  username: undefined,
+  avatar: null,
+  id: undefined,
+  token: undefined,
+};
+
 export default new Vuex.Store({
-  state: {
-    isAuthenticated: false,
-    user: undefined,
+  state: state,
+  getters: {
+    username: (state) => state.username,
+    isAuthenticated: (state) => state.isAuthenticated,
+    avatar: (state) => state.avatar,
+    id: (state) => state.id,
+    token: (state) => state.token,
   },
-  getters: {},
   mutations: {
-    login(state, loginDto) {
-      state.user = loginDto;
+    login(state, payload) {
+      state.username = payload.username;
+      state.id = payload.id;
       state.isAuthenticated = true;
+      state.token = payload.token;
+    },
+  },
+  actions: {
+    async getAvatar(context) {
+      try {
+        const id = context.state.id;
+        if (id === undefined) return undefined;
+        const response = await axios.get("/users/" + id + "/avatar", {
+          responseType: "blob",
+        });
+        if (response.status === 204) return;
+        const blob = new Blob([response.data]);
+        context.state.avatar = URL.createObjectURL(blob);
+      } catch (e) {
+        console.error(e);
+        context.state.avatar = undefined;
```

Comment:
I do not think that it should be refactored here. The avatar of the current user is stored in the global store because it is constantly used. I do not think that reproducing a database locally is a good idea. But if the goal is to produce a function utility that is in ```another file``` and used here as well as when avatars are fetched for other users, I think it is a good idea. I opened  #116